sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/SecretClient.java

```diff
@@ -220,13 +258,19 @@ public KeyVaultSecret getSecret(String name, String version) {
      * to calling {@link #getSecret(String)}, with the latest version being retrieved.
      * @param context Additional context that is passed through the HTTP pipeline during the service call.
      * @return A {@link Response} whose {@link Response#getValue() value} contains the requested {@link KeyVaultSecret}.
+     * @throws NullPointerException if {@code name} is {@code null}.
      * @throws ResourceNotFoundException when a secret with {@code name} and {@code version} doesn't exist in the key
      * vault.
-     * @throws HttpResponseException if {@code name} or {@code version} is empty string.
+     * @throws HttpResponseException if {@code name} and {@code version} is empty string.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<KeyVaultSecret> getSecretWithResponse(String name, String version, Context context) {
-        return implClient.getSecretWithResponse(name, version, context);
+        Objects.requireNonNull(name, "Secret name cannot be null.");
```

Comment:
@JonathanGiles, @srnagar, @lmolkova this is a new exception being thrown checking that the `name` is non-null. While this is a breaking change as this adds a new exception being thrown this fixes a larger issue where the REST API paths used by KeyVault here is `GET {keyVaultUrl}/secrets/{secretName}/{secretVersion}` for getting a single secret and `GET {keyVaultUrl}/secrets` for listing secrets. This leaves a possibility when name and version is null or empty string for the final URL to become `GET {keyVaultUrl}/secrets//` which ends up calling into the listing resulting in a deserialization exception as the response type is different in that API.

FYI @vcolin7

Reply:
Should we even consider changing this to check for it not to be either `null` nor empty? Passing an empty string value will also try to list all secrets.

Reply:
Reverting this change to have it be done in another PR where it can be discussed more directly.